File: Week_08/G20200343030436/387.字符串中的第一个唯一字符.java
import java.util.HashMap;

/*
 * @lc app=leetcode.cn id=387 lang=java
 *
 * [387] 字符串中的第一个唯一字符
 */

// @lc code=start
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> count = new HashMap<Character, Integer>();
        int len = s.length();
        for (int i = 0; i < len; i++) {
            char temp = s.charAt(i);
            count.put(temp, count.getOrDefault(temp, 0) + 1);
        }

        for (int i = 0; i < len; i++) {
            if (count.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        return -1;
    }
}
// @lc code=end

File: Week_08/G20200447010018/387.字符串中的第一个唯一字符.js
/*
 * @lc app=leetcode.cn id=387 lang=javascript
 *
 * [387] 字符串中的第一个唯一字符
 *
 * https://leetcode-cn.com/problems/first-unique-character-in-a-string/description/
 *
 * algorithms
 * Easy (43.21%)
 * Likes:    197
 * Dislikes: 0
 * Total Accepted:    69.2K
 * Total Submissions: 156.6K
 * Testcase Example:  '"leetcode"'
 *
 * 给定一个字符串，找到它的第一个不重复的字符，并返回它的索引。如果不存在，则返回 -1。
 * 
 * 案例:
 * 
 * 
 * s = "leetcode"
 * 返回 0.
 * 
 * s = "loveleetcode",
 * 返回 2.
 * 
 * 
 * 
 * 
 * 注意事项：您可以假定该字符串只包含小写字母。
 * 
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let res = -1;
    for (let i = 0; i < s.length; i++) {
        if (i === s.indexOf(s[i]) && s.indexOf(s[i], i + 1) === -1) {
            res = i;
            break;
        }
    }
    return res;
};
// @lc code=end

